database/database/connection.py

```diff
@@ -515,3 +515,552 @@ def get_latest_products_with_unknown_sustainability_label(self) -> pd.DataFrame:
+                db_session.query(
+                    get_sustainability_attributes.c.prod_id,
+                    func.max(get_sustainability_attributes.c.eco_chemicals).label("eco_chemicals"),
+                    func.max(get_sustainability_attributes.c.eco_lifetime).label("eco_lifetime"),
+                    func.max(get_sustainability_attributes.c.eco_water).label("eco_water"),
+                    func.max(get_sustainability_attributes.c.eco_inputs).label("eco_inputs"),
+                    func.max(get_sustainability_attributes.c.eco_quality).label("eco_quality"),
+                    func.max(get_sustainability_attributes.c.eco_energy).label("eco_energy"),
+                    func.max(get_sustainability_attributes.c.eco_waste_air).label("eco_waste_air"),
+                    func.max(get_sustainability_attributes.c.eco_environmental_management).label(
+                        "eco_environmental_management"
+                    ),
+                    func.max(get_sustainability_attributes.c.social_labour_rights).label(
+                        "social_labour_rights"
+                    ),
+                    func.max(get_sustainability_attributes.c.social_business_practice).label(
+                        "social_business_practice"
+                    ),
+                    func.max(get_sustainability_attributes.c.social_social_rights).label(
+                        "social_social_rights"
+                    ),
+                    func.max(get_sustainability_attributes.c.social_company_responsibility).label(
+                        "social_company_responsibility"
+                    ),
+                    func.max(get_sustainability_attributes.c.social_conflict_minerals).label(
+                        "social_conflict_minerals"
+                    ),
+                    func.round(func.avg(get_sustainability_attributes.c.cred_credibility)).label(
+                        "mean_credibility"
+                    ),
+                )
+                .group_by(get_sustainability_attributes.c.prod_id)
+                .subquery()
+            )
+
+            get_eco_and_social_means = (
+                db_session.query(
+                    get_max_sustainability_scores.c.prod_id,
+                    (
+                        func.sum(
+                            get_max_sustainability_scores.c.eco_chemicals
+                            + get_max_sustainability_scores.c.eco_lifetime
+                            + get_max_sustainability_scores.c.eco_water
+                            + get_max_sustainability_scores.c.eco_inputs
+                            + get_max_sustainability_scores.c.eco_quality
+                            + get_max_sustainability_scores.c.eco_energy
+                            + get_max_sustainability_scores.c.eco_waste_air
+                            + get_max_sustainability_scores.c.eco_environmental_management
+                        )
+                        / 8
```

Comment:
could this be computed with `func.avg()`?
Or otherwise it's always nice to have constants like the 8 and the 5 below defined as variables, something like
N_ECO_DIMENSIONS = 8
N_SOC_DIMENSIONS = 5

Reply:
I believe is not possible to use func.avg() over different columns in the same row, so I added constants as suggested. Fixed in [54e1fd0](https://github.com/calgo-lab/green-db/pull/103/commits/54e1fd09bb7ea35f7b444c2cd26b3fadd955ecb1)